runtimes/core/src/log/writers.rs

```diff
@@ -53,25 +54,63 @@ impl ActorWriter {
     pub fn new<W: Write + Sync + Send + 'static>(mut writer: W) -> Self {
         let (sender, recv) = mpsc::sync_channel::<Vec<u8>>(10_000);
         std::thread::spawn(move || {
-            while let Ok(bytes) = recv.recv() {
-                Self::write_with_retry(&mut writer, &bytes);
+            while let Ok(bytes) = Self::recv_batch(&recv) {
+                Self::write_batch_with_retry(&mut writer, &bytes);
             }
         });
         Self { sender }
     }
 
-    fn write_with_retry<W: Write>(writer: &mut W, bytes: &[u8]) {
+    fn recv_batch(recv: &Receiver<Vec<u8>>) -> Result<Vec<Vec<u8>>, RecvError> {
+        const MAX_BATCH_SIZE: usize = 256;
+        let mut bufs = Vec::with_capacity(MAX_BATCH_SIZE);
```

Comment:
I would make `Vec::new()` instead; the most common scenario is that we'll get a single log message so this will over-allocate